gno.land/pkg/gnoland/types.go

```diff
@@ -19,7 +19,92 @@ func ProtoGnoAccount() std.Account {
 	return &GnoAccount{}
 }
 
+// GnoGenesis defines the gno genesis API,
+// adopted by differing genesis state implementations
+type GnoGenesis interface {
+	// GenesisBalances returns the genesis balances associated
+	// with the Gno genesis state
+	GenesisBalances() []Balance
+
+	// GenesisTxs returns the genesis transactions associated
+	// with the Gno genesis state
+	GenesisTxs() []GenesisTx
+}
+
+type GenesisTx interface {
+	// Tx returns the standard TM2 transaction
+	Tx() std.Tx
+
+	// Metadata returns the metadata tied
+	// to the tx, if any
+	Metadata() *GenesisTxMetadata
+}
+
 type GnoGenesisState struct {
 	Balances []Balance `json:"balances"`
 	Txs      []std.Tx  `json:"txs"`
 }
+
+func (g GnoGenesisState) GenesisBalances() []Balance {
+	return g.Balances
+}
+
+func (g GnoGenesisState) GenesisTxs() []GenesisTx {
+	genesisTxs := make([]GenesisTx, len(g.Txs))
+
+	for i, tx := range g.Txs {
+		genesisTxs[i] = gnoGenesisTx{
+			tx: tx,
+		}
+	}
+
+	return genesisTxs
+}
+
+type gnoGenesisTx struct {
+	tx std.Tx
+}
+
+func (g gnoGenesisTx) Tx() std.Tx {
+	return g.tx
+}
+
+func (g gnoGenesisTx) Metadata() *GenesisTxMetadata {
+	return nil
+}
+
+type MetadataGenesisState struct {
+	Balances []Balance    `json:"balances"`
+	Txs      []MetadataTx `json:"txs"`
+}
+
+type MetadataTx struct {
+	GenesisTx  std.Tx            `json:"tx"`
+	TxMetadata GenesisTxMetadata `json:"metadata"`
+}
```

Comment:
nit: MetadataTx = Metadata + Tx = 

```suggestion
type MetadataTx struct {
	Metadata GenesisTxMetadata `json:"metadata"`
	Tx  std.Tx            `json:"tx"`
}
```

Reply:
How I see you didn't do it because it conflict with the `Tx()` method.